src/index.js

```diff
@@ -130,12 +183,43 @@ const validate = async (publicKey, entry) => {
   log(`ipns entry for ${value} is valid`)
 }
 
+/**
+ * @param {IPNSEntry} entry
+ */
+const validateCborDataMatchesPbData = async (entry) => {
+  if (!entry.data) {
+    throw errCode(new Error('Record data is missing'), ERRORS.ERR_INVALID_RECORD_DATA)
+  }
+
+  const data = cborg.decode(entry.data)
```

Comment:
```suggestion
  const data = cborg.decode(entry.data)
  if (Number.isInteger(data.sequence)) {
    // sequence must be a BigInt, but DAG-CBOR doesn't preserve this for Numbers within the safe-integer range
    data.sequence = BigInt(data.sequence)
  }
```

as per discussion @ https://github.com/rvagg/cborg/pull/18#issuecomment-855574393

`Number.isInteger()` is nice here because it'll only pass _if_ it's both a `Number` and doesn't have a fractional comonent so it'll safely convert to `BigInt` and anything else will fail the `data.sequence !== entry.sequence` test below.